build-scripts/ubuntu-1604/build-3rd-parties.sh

```diff
@@ -79,6 +79,9 @@ function build_from_pypi {
 
 # TODO duplicates list from Jenkinsfile.cd
 
+SCRIPT_PATH="${BASH_SOURCE[0]}"
+cd `dirname ${SCRIPT_PATH}`
```

Comment:
I'd recommend switching to a `push` `pop` approach to changing directories.  Simply changing directories in the middle of a build process can have unexpected consequences that are difficult to debug.  The `push` `pop` approach ensures the working directory remains unchanged for all subsequent processes.

Example of `push` `pop` approach:
```
pushd ${SCRIPT_PATH} >/dev/null

# Do some work in ${SCRIPT_PATH} 

popd >/dev/null
```

Reply:
Done